### ISSUE TITLE
Code Fix for xUnit-2004 produces invalid code

### ISSUE THREAD

**Issue by sys27** [xunit/xunit#1468]
Visual Studio reports Warning: xUnit-2004 - Do not use Assert.Equal() to check for boolean conditions. And suggests a fix, but fix produces invalid C# code.

**Code to reproduce:**
`Assert.Equal(true, (object)null);`
or
`Assert.Equal(true, anyMethodWhichReturnsObject());`

**Actual Behavior:**
`Assert.True((object)null);`
Error: cannot convert from 'object' to 'bool'.

**Expected Behavior:**
`Assert.True((bool)anyMethodWhichReturnsObject());`
or don't show code fix.

**Comment by stakx** [comment]
I have encountered this problem, too (e.g. [here](https://github.com/moq/moq4/blob/0633af99fe5f2fbb6c8a9844244f7dbfd2404493/UnitTests/MockedEventsFixture.cs#L184-L207)).

>**Expected Behavior:**
>`Assert.True((bool)anyMethodWhichReturnsObject());`

I respectfully disagree. Consider this test:

>```csharp
>object obj = "unexpected value";
>Assert.True((bool)obj, "custom message");
>```

In this case, the user message (`custom message`) will be lost. Instead, the test run will log the message `System.InvalidCastException : Specified cast is not valid.`.

>or don't show code fix.

I believe this would be the correct behaviour. This would mean that the test I just showed could be legally written as follows:

>```csharp
>object obj = "unexpected value";
>Assert.Equal(true, "custom message");
>```

This way, the user message will get logged correctly in the test run output.
